Lib/http/server.py

```diff
@@ -95,12 +95,15 @@
 import io
 import mimetypes
 import os
+import gzip
+import zlib
```

Comment:
In a7f1f47 I put these imports in a `try` block and set a flag `has_gzip` ; if Python is built without these modules, compression is ignored.